components/asp/assembler/TraceAssembler.py

```
@@ -0,0 +1,168 @@
+                return trace
+
+        # TODO: If we reached this point, the trace did not end. Add 
+        # functionality to deal with this case.
+        return trace
+        
+    def findLink(self, node):
+        '''
+            Find the linked node.
+        '''
+        query = f'SELECT * FROM "IOEVENTS" WHERE ("type" = ? or "type" = ?)\
+              and "adli_execution_id" = ? and "adli_execution_index" = ?' 
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)